### ISSUE TITLE
Important feature request - hook into tsc --watch

### ISSUE THREAD

**Issue by ORESoftware** [Microsoft/TypeScript#16602]
**TypeScript Version:**  2.4.0 / nightly (2.5.0-dev.201xxxxx)

I have been asking about this feature for almost a year (I think) - I have several use cases where I need to hook into a `tsc -w ` process, so that I take action only after the latest transpilation routine has completed. `tsc -w` is much quicker than transpiling from scratch everytime, otherwise I would do the latter.

I tried creating my own solution using the programmatic API, but I was thwarted:
https://github.com/Microsoft/TypeScript/issues/14722

I could **really** use a way to hook into tsc -w, so as to avoid race conditions, and only take action after it's done transpiling.

I am not sure how it would work? Websockets? Probably not necessary. Lockfiles on the filesystem? Maybe.

**Comment by Hotell** [comment]
you may wanna use this until/if this will be/is implemented https://github.com/gilamran/tsc-watch

**Comment by ORESoftware** [comment]
@Hotell thanks - I see there is 1 issue (the first ever issue) on that repo that hasn't been responded to in 11 days, pretty sad.

**Comment by ORESoftware** [comment]
@DanielRosenwasser - any news on how to hook into tsc --watch?

I see that the Angular `ng test` command will hook into `webpack --watch` and I am hoping to do something similar with `tsc`.

**Comment by ORESoftware** [comment]
I am closing this in favor of a new issue. Thanks.
